eventuals/generator.h

```diff
@@ -80,27 +75,31 @@ struct HeapGenerator {
     Callback<std::exception_ptr>* fail_;
     Callback<>* stop_;
     std::conditional_t<
-        std::is_void_v<Value_>,
+        std::is_void_v<Arg_>,
         Callback<>,
-        Callback<Value_>>* body_;
+        Callback<Arg_>>* body_;
     Callback<>* ended_;
   };
 
   HeapGenerator(E_ e)
     : adapted_(
-        std::move(e).template k<void>(
-            Adaptor<Value_>{&begin_, &fail_, &stop_, &body_, &ended_})) {}
+        std::move(e).template k<From_>(
+            Adaptor<To_>{&begin_, &fail_, &stop_, &body_, &ended_})) {}
```

Comment:
What about the plan to just drop `To_` here and have `Adaptor` which uses the types from `HeapGenerator`?